src/strategy/immediate.rs

```diff
@@ -34,37 +26,19 @@ impl StrategyImmediate {
         let immediate = future::ok(true);
         Box::pin(immediate)
     }
-
-    pub(crate) fn can_check_and_fetch(&self) -> Pin<Box<dyn Future<Output = Result<bool, Error>>>> {
-        trace!("immediate strategy, can check updates: {}", self.check);
-
-        let immediate = future::ok(self.check);
-        Box::pin(immediate)
-    }
 }
 
 impl Default for StrategyImmediate {
     fn default() -> Self {
-        Self {
-            check: true,
-            finalize: true,
-        }
+        Self {}
```

Comment:
You can now drop this `impl Default` block and directly put a `#[derive(Default)]` annotation on the struct.